### PR TITLE
Update start block to current block if checkpoint is 0

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- Updated chaindatafetcher start block to current block if checkpoint is 0.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 1]
```diff
@@ -317,7 +317,12 @@ func (f *ChainDataFetcher) setCheckpoint() {
 	if err != nil {
 		logger.Crit("ReadCheckpoint is failed", "err", err)
 	}
+
+	if checkpoint == 0 {
+		checkpoint = f.blockchain.CurrentHeader().Number.Int64()
+	}
 	f.checkpoint = checkpoint
+	logger.Info("Chaindatafetcher checkpoint is set", "checkpoint", f.checkpoint)
```
Comment (KimKyungup) [inline datasync/chaindatafetcher/chaindata_fetcher.go:325]:
```suggestion
	logger.Info("Chaindatafetcher initial checkpoint is set", "checkpoint", f.checkpoint)
```

**Comment by ehnuje** [comment]
Is this to catch up the latest blocks first and then load the past blocks later?

File: datasync/chaindatafetcher/chaindata_fetcher.go (hunk 1 above)
Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/chaindata_fetcher.go]: Reflected.

**Comment by jeongkyun-oh** [comment]
> Is this to catch up the latest blocks first and then load the past blocks later?

@ehnuje If the chaindatafetcher doesn't have a checkpoint, it starts pushing data from the current block, not from the first block. We assume that the user needs data from the current block initially.

**Review by aidan-kwon — APPROVED** [review]

**Review by winnie-byun — APPROVED** [review]

**Review by yoomee1313 — APPROVED** [review]

**Review by KimKyungup — APPROVED** [review]

**Comment by jeongkyun-oh** [comment]
@ehnuje Please take a look.

**Pull request merged**